src/Testing/ZopeTestCase/zopedoctest/functional.py

```diff
@@ -237,8 +237,9 @@ def setup_test_class(self):
 
         # If the test_class does not have a runTest method, we add
         # a dummy attribute so that TestCase construction works.
+        # runTest may get called, so set it to something that works.
         if not hasattr(test_class, 'runTest'):
-            setattr(test_class, 'runTest', None)
+            setattr(test_class, 'runTest', lambda x: None)
 
```

Comment:
What I mean is a module level test_suite function, for example https://github.com/zopefoundation/Zope/blob/master/src/Products/Five/browser/tests/test_pages.py#L69

The double under main function isn't required anymore. And if a test module has only test classes inherited from unittest.TestCase one can also leave out the test_suite function. But if there's old style tests around, I think one still needs the extra function.